### PR TITLE
Reimplement action routing

### PR DESCRIPTION
**Pull request by ketzusaka**
The original implementation for this got lost somewhere. This adds it back in with tests.

### CODE REVIEW

File: Sources/Vapor/Routing/Application+Route.swift [hunk 1]
```diff
@@ -102,6 +102,75 @@ extension Application {
...
     /**
+        Adds a route handled by a type that can be initialized with an `Application`.
+        This method is useful if you have a controller and would like to add an action
+        that is not a common REST action.
+
+        Here's an example of how you would add a route with this method:
+
+        ```app.add(.get, path: "/foo", action: TestController.foo)```
+
+        - parameter     method:     The `Request.Method` that the action should be executed for.
```
Comment (tannernelson) [inline Sources/Vapor/Routing/Application+Route.swift]: I think the rest of the project uses one space between words. If so, let's keep it consistent.

**Comment by tannernelson** [comment]
Looks good 👍

File: Sources/Vapor/Routing/Application+Route.swift (hunk 1 above)
Comment (ketzusaka) [inline Sources/Vapor/Routing/Application+Route.swift]: Yeah, it is. Updated!

**Comment by tannernelson** [comment]
Sweet, thanks!

**Pull request merged**